packages/terra-tag/src/TagsUtils.jsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * Takes tag as input and returns its width.
+ */
+const getTagWidth = (tag) => {
+  const tagWidth = tag.offsetWidth + parseFloat(window.getComputedStyle(tag, null).getPropertyValue('margin-right'), 10);
+  return tagWidth;
+};
+
+/**
+ * Takes rollUpTag as input and returns its width.
+ */
+const getRollUpTagWidth = (rollUpTag) => {
+  const rollupTagWidth = rollUpTag.parentElement.offsetWidth + parseFloat(window.getComputedStyle(rollUpTag.parentElement, null).getPropertyValue('margin-right'), 10);
+  return rollupTagWidth;
+};
+
+/**
+ * Takes tag container ref as input and returns the index value from which the tags need to be hidden/rolled up.
+ */
+const getRollUpIndex = (tagListRef) => {
+  const rollUpTag = tagListRef.current.querySelector('[data-terra-rollup-tag]');
+  const tags = tagListRef.current.querySelectorAll('[data-terra-tag]');
```

Comment:
Why do you need to query the DOM?  Don't you have the reference data in React props?

Reply:
tagListRef is a ref item prop, which contains all tag nodes plus rollup tag node. For differentiating we use data prop here. Thanks

Reply:
We should not have to access the DOM for this information though.  Somehow we populated the DOM from the data passed to the component.  React should rely on the virtual DOM, when possible.